velox/functions/prestosql/ArrayFunctions.h

```diff
@@ -418,4 +418,51 @@ struct ArrayFrequencyFunction {
   }
 };
 
+template <typename TExecParams, typename T>
+struct ArrayNormalizeFunction {
+  VELOX_DEFINE_FUNCTION_TYPES(TExecParams);
+
+  FOLLY_ALWAYS_INLINE void callNullFree(
+      out_type<velox::Array<T>>& result,
+      const null_free_arg_type<velox::Array<T>>& inputArray,
+      const null_free_arg_type<T> p) {
+    VELOX_USER_CHECK_GE(
+        p, 0, "array_normalize only supports non-negative p: {}", p);
+
+    // If the input array is empty, then the empty result should be returned
+    // similar to Presto.
+    if (inputArray.size() == 0) {
+      return;
+    }
+
+    result.reserve(inputArray.size());
+
+    // If p = 0, then it is L0 norm. Presto version returns the input array.
```

Comment:
Would you clarify this in the documentation as well?

Reply:
Updated.